pkg/utils/requester_test.go

```diff
@@ -190,12 +190,36 @@ func TestPostObj(t *testing.T) {
 	assert.NotNil(t, err)
 }
 
+// mockLogger exists to check that the logged error is what we expected
+type mockLogger struct {
+	Errors []error
+}
+
+func (t *mockLogger) Debug(message string)   {}
+func (t *mockLogger) Info(message string)    {}
+func (t *mockLogger) Warning(message string) {}
+func (t *mockLogger) Error(message string, err interface{}) {
+	t.Errors = append(t.Errors, err.(error))
```

Comment:
It can cause panic in two cases:
1. when err interface{} is nil 
2. when err interface{} is not error type

We can avoid those by following way:
`
if _, ok := err.(error); ok {
      t.Errors = append(t.Errors, err)
}
`
What do you think?

Reply:
Sounds good - added this check to `Error()`.

Also added an early failure in the dependent test function in the case where we don't have the expected error.